src/main/java/com/box/sdk/BoxAPIResponse.java

```diff
@@ -134,19 +134,21 @@ public InputStream getBody(ProgressListener listener) {
      */
     public void disconnect() {
         try {
-            if (this.rawInputStream == null) {
+            if (this.rawInputStream == null && this.connection != null) {
```

Comment:
Sure, if you think that's good then let's go with that. I just wanted to be a little cautious considering that there was also a change to `this.inputStream` underneath.